cdap-data-fabric/src/main/java/io/cdap/cdap/spi/data/sql/PostgreSqlStructuredTable.java

```diff
@@ -1026,6 +1061,18 @@ private String getCountStatement(Collection<Range> ranges) {
         appendRange(statement, range);
       }
     }
+
+    if (!filterIndexes.isEmpty()) {
```

Comment:
Cant we just add filterindexes first and then add range. In that case we do not need to manipulate `()` position

Reply:
Discussed offline...

Reply:
Adding how the query looks like:

```
SELECT COUNT(*) FROM application_specs WHERE ((namespace)<('system') OR (namespace)>('system')) AND (latest is NULL OR latest = 'TRUE')
```